tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -0,0 +1,185 @@
+          : Color.fromHex(department.color)
+            .darken(90)
+            .toString(),
+        'font-size': '1.1rem',
+        'cursor': job.unavailable_reason ? 'initial' : 'pointer',
+      }}
+      tooltip={
+        job.unavailable_reason ||
+        (job.prioritized ? (
+          <>
+            <p style={{ 'margin-top': '0px' }}>
+              <b>The HoP wants more people in this job!</b>
+            </p>
+            {job.description}
+          </>
+        ) : (
+          job.description
+        ))
+      }
+      onClick={() => {
+        !job.unavailable_reason && data.onClick();
+      }}>
+      <>
+        {jobIcon && <Icon name={jobIcon} />}
+        {job.command ? <b>{jobName}</b> : jobName}
+        <span
+          style={{
+            'white-space': 'nowrap',
+            'position': 'absolute',
+            'right': '0.5em',
+          }}>
+          {job.used_slots} / {job.open_slots}
+        </span>
+      </>
+    </Button>
+  );
+};
+
+export const JobSelection = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  if (!data?.departments_static) {
+    return null; // Stop TGUI whitescreens with TGUI-dev!
+  }
+  const departments: Record<string, Department> = deepMerge(
+    data.departments,
+    data.departments_static
+  ); // Why the fuck is it so hard to clone objects properly in JS?!
+
+  return (
+    <Window width={1012} height={666 /* Hahahahahaha */}>
+      <Window.Content scrollable>
+        <StyleableSection
+          title={
+            <Button
+              style={{ 'position': 'absolute', 'right': '1em' }}
+              onClick={() => act('SelectedJob', { 'job': 'Random' })}
+              content="Random Job!"
+              tooltip="Roll target random job. You can re-roll or cancel your random job if you don't like it."
+            />
+          }
+          titleStyle={{ 'min-height': '3.4em' }}>
+          <Box wrap="wrap" style={{ 'columns': '20em' }}>
+            {Object.entries(departments).map((departmentEntry) => {
+              const departmentName = departmentEntry[0];
+              const entry = departmentEntry[1];
+              return (
+                <Box key={departmentName} minWidth="30%">
+                  <StyleableSection
+                    title={
+                      <>
+                        {departmentName}
+                        <span
+                          style={{
+                            'font-size': '1rem',
+                            'white-space': 'nowrap',
+                            'position': 'absolute',
+                            'right': '1em',
+                            'color': Color.fromHex(entry.color)
+                              .darken(60)
+                              .toString(),
+                          }}>
+                          {entry.open_slots +
+                            (entry.open_slots === 1 ? ' Slot' : ' Slots') +
+                            ' Available'}
```

Comment:
```suggestion
                            (entry.open_slots === 1 ? ' slot' : ' slots') +
                            ' available'}
```

Nit, I just think this looks better